File: C/Algorithm/Algorithm I/167. Two Sum II - Input Array Is Sorted.c
int* twoSum(int* numbers, int numbersSize, int target, int* returnSize) {
    int *indicesSolution = malloc(2 * sizeof(int));
    
    int leftPointer = 0;
    int rightPointer = numbersSize - 1;
    int currentSum = numbers[leftPointer] + numbers[rightPointer];

    while (currentSum != target) {
        if (currentSum > target)
            rightPointer--;
        else
            leftPointer++;

        currentSum = numbers[leftPointer] + numbers[rightPointer];
    }

    indicesSolution[0] = leftPointer + 1;
    indicesSolution[1] = rightPointer + 1;
    *returnSize = 2;

    return indicesSolution;
}

File: TypeScript/Algorithm/Algorithm I/167. Two Sum II - Input Array Is Sorted.ts
/**
 * Finds a pair of numbers in a sorted array that add up to the given target.
 * @param numbers - A sorted array of numbers.
 * @param target - The target sum to be achieved.
 * @returns An array containing the indices of the two numbers that add up to the target.
 */
function twoSum(numbers: number[], target: number): number[] {
    const numbersLength: number = numbers.length;

    let leftPointer: number = 0;
    let rightPointer: number = numbersLength - 1;

    while (leftPointer < rightPointer) {
        const currentSum: number = numbers[leftPointer] + numbers[rightPointer];

        if (currentSum === target)
            return [leftPointer + 1, rightPointer + 1];

        else if (currentSum > target)
            rightPointer--;
    
        else
            leftPointer++;
    }

    return [leftPointer + 1, rightPointer + 1];
};
